src/platform/linux/x11/mod.rs

```diff
@@ -442,10 +443,16 @@ impl EventsLoop {
                     }
                     ffi::XI_FocusIn => {
                         let xev: &ffi::XIFocusInEvent = unsafe { &*(xev.data as *const _) };
+                        let mut windows = self.windows.lock().unwrap();
```

Comment:
I don't think we want to hold this lock across a `callback` invocation, in case the callback calls back into winit code which tries to lock it in turn. Rust mutexes are not (guaranteed to be?) recursive, so this would lead to breakage.